### PR TITLE
fix(mongo-datasource): should return record with empty flattened record

### PR DESCRIPTION
**Pull request by Scra3**
linked to: CU-86bxzpqjc
### General

- [ ] Write an explicit title for the Pull Request, following [Conventional Commits specification](https://www.conventionalcommits.org)
- [ ] Test manually the implemented changes
- [ ] Validate the code quality (indentation, syntax, style, simplicity, readability)

### Security

- [ ] Consider the security impact of the changes made

### CODE REVIEW

File: packages/datasource-mongoose/src/utils/pipeline/condition-generator.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import { AnyExpression } from 'mongoose';
+
+export const RECORD_DOES_NOT_EXIST = 'FOREST_RECORD_DOES_NOT_EXIST';
+export default class ConditionGenerator {
+  static tagRecordIfNotExist(field: string, then: AnyExpression): AnyExpression {
```
Comment (realSpok) [inline packages/datasource-mongoose/src/utils/pipeline/condition-generator.ts:5]: those 2 look similar. Do you think you can factorize ?

File: packages/datasource-mongoose/src/utils/pipeline/condition-generator.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import { AnyExpression } from 'mongoose';
+
+export const RECORD_DOES_NOT_EXIST = 'FOREST_RECORD_DOES_NOT_EXIST';
```
Comment (realSpok) [inline packages/datasource-mongoose/src/utils/pipeline/condition-generator.ts:3]:
```suggestion
export const FOREST_RECORD_DOES_NOT_EXIST = 'FOREST_RECORD_DOES_NOT_EXIST';
```

File: packages/datasource-mongoose/src/utils/pipeline/reparent.ts [hunk 3]
```diff
@@ -41,11 +42,11 @@ export default class ReparentGenerator {
           newRoot: {
```
Comment (realSpok) [inline packages/datasource-mongoose/src/utils/pipeline/reparent.ts:42]: Those 2 functions as well, but you only modified them…

File: packages/datasource-mongoose/src/utils/pipeline/condition-generator.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
+import { AnyExpression } from 'mongoose';
```
Comment (realSpok) [inline packages/datasource-mongoose/src/utils/pipeline/condition-generator.ts:1]: Do you think we should leave a comment explaining what is going on here, especially why we are doing it ? Today it is clear to us but in the future…

File: packages/datasource-mongoose/src/collection.ts [hunk 5]
```diff
@@ -57,30 +56,21 @@ export default class MongooseCollection extends BaseCollection {
...
-  ): Promise<RecordData[]> {
+  private async _list(filter: PaginatedFilter, projection: Projection): Promise<RecordData[]> {
     const lookupProjection = projection.union(
       filter.conditionTree?.projection,
       filter.sort?.projection,
     );
 
-    const records = await this.model.aggregate([
-      ...(pipelineBefore || []),
+    const pipeline = [
```
[the remark this answers is not in the document]
Comment (Scra3) [reply packages/datasource-mongoose/src/collection.ts:68]: Yeah, and it helps a lot to debug with breakpoints

**Review by realSpok — APPROVED** [review]
- ✅  Code review
  - ✅  Use pure functions when possible
  - ✅  Performance concerns
  - ✅  Security concerns
  - ✅  The PR tackle one subject only
- ✅ Automatic tests
  - ✅  Unit tests
  - ✅  Integration tests
- ✅ Manual tests
  -  ✅  Test the functionality
  -  ✅  Test error cases
- ✅  PR title
- ✅  Destination branch
- ✅  PR linked to the clickup task

**Pull request merged**